### PR TITLE
Fixes learning rate and momentum definitions

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes learning rate and momentum definitions for linear regression.

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Naarcha-AWS — APPROVED** [review]

File: _ml-commons-plugin/algorithms.md [hunk 1]
```diff
@@ -65,8 +65,8 @@ Linear regression maps the linear relationship between inputs and outputs. In ML
 
 Parameter | Type   | Description | Default Value
 :--- |:--- | :--- | :---
-learningRate | Double | The rate of speed at which the gradient moves during descent | 0.01
-momentumFactor | Double | The medium-term from which the regressor rises or falls | 0
+learningRate | Double | The gradient descent step size at each iteration when moving toward a minimum of a loss function or optimal value. | 0.01
```
Comment (alicejw-aws) [inline _ml-commons-plugin/algorithms.md:68]:
```suggestion
learningRate | Double |  The step size at each iteration that the gradient descends when moving toward a minimum loss of function or optimal value. | 0.01
```

Comment (alicejw-aws) [reply _ml-commons-plugin/algorithms.md:68]: was a bit confused sentence order. suggested clarification

**Review by alicejw-aws — APPROVED** [review]
looks good. made a suggestion

File: _ml-commons-plugin/algorithms.md [hunk 2]
```diff
@@ -65,8 +65,8 @@ Linear regression maps the linear relationship between inputs and outputs. In ML
...
+momentumFactor | Double | The extra weight factors that accelerate the rate at which the weight is adjusted. This helps move the minimization routine out of local minima.  | 0
 epsilon | Double | The criteria used to identify a linear model | 1.00E-06
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/algorithms.md:70]: @Zhangxunmt @ylwu-amzn: Could we copy the definitions for epsilon, beta1, beta2, and decayRate from the Logistic Regression section so they are as follows:
| epsilon | Double | The value for stabilizing gradient inversion | 0.1 |
| beta1 | Double | The exponential decay rates for the moment estimates | 0.9 |
| beta2 | Double | The exponential decay rates for the moment estimates | 0.99 |
| decayRate | Double | The Root Mean Squared Propagation (RMSProp) | 0.9 |

Or do these not mean the same thing in this context? Thanks!

Comment (kolchfa-aws) [reply _ml-commons-plugin/algorithms.md:70]: Sure, thank you @Zhangxunmt!

**Comment by ylwu-amzn** [comment]
Thanks @greaa-aws 

@kolchfa-aws , thanks for improving the tech doc. How about also tuning the `learingRate` based on comment from @greaa-aws

File: _ml-commons-plugin/algorithms.md (hunk 2 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/algorithms.md]: @greaa-aws: thank you for following up! Please feel free to suggest other documentation improvements so we can implement them.

**Comment by kolchfa-aws** [comment]
@ylwu-amzn @greaa-aws I changed the definition for the `learningRate` to read "The initial step size used in an iterative optimization algorithm." in both instances (PR #1546).

**Pull request merged**